core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -101,6 +101,7 @@ class GroupCoordinator(val brokerId: Int,
 
   def handleJoinGroup(groupId: String,
                       memberId: String,
+                      groupInstanceId: String,
```

Comment:
I think it would be clearer if we represented this as `Option[String]`.

Reply:
I think we handle the null case when building the join group request struct?

Reply:
I'm just saying that it would be clearer to represent the difference between static and dynamic members by using an optional field. Otherwise you have to dig into the code to make sure the uses are all safe. For example, we have a bunch of cases below where we are using the empty instance id in calls to `hasStaticMember`. This opens the door to bugs if we are not really careful with our checking. The nice thing about options is that they force us to check for absence.